File: apps/mobile/lib/core/phone.dart
String _normalizeTurkishPhone(String raw) {
  final digits = raw.replaceAll(RegExp(r'\D'), '');
  var local = digits;
  if (local.startsWith('90') && local.length > 10) local = local.substring(2);
  if (local.startsWith('0')) local = local.substring(1);
  return local;
}

/// Görüntüleme için: "0555 *** ** 22" — sayfa/ekran kaynağında numarayı açık bırakmaz.
String maskPhone(String raw) {
  final local = _normalizeTurkishPhone(raw);
  if (local.length != 10) return raw;
  final first3 = local.substring(0, 3);
  final last2 = local.substring(8, 10);
  return '0$first3 *** ** $last2';
}

/// wa.me linki için ülke koduyla birlikte boşluksuz uluslararası format.
String toWhatsAppPhone(String raw) {
  final local = _normalizeTurkishPhone(raw);
  return '90$local';
}

File: apps/web/lib/phone.ts
/** "+90 555 123 45 22", "05551234522", "5551234522" gibi girdileri 10 haneli yerel numaraya indirger. */
function normalizeTurkishPhone(raw: string): string {
  const digits = raw.replace(/\D/g, "");
  let local = digits;
  if (local.startsWith("90") && local.length > 10) local = local.slice(2);
  if (local.startsWith("0")) local = local.slice(1);
  return local;
}

/** Görüntüleme için: "0555 *** ** 22" — ortadaki haneleri maskeler, numarayı sayfa kaynağında açık bırakmaz. */
export function maskPhone(raw: string): string {
  const local = normalizeTurkishPhone(raw);
  if (local.length !== 10) return raw;
  const first3 = local.slice(0, 3);
  const last2 = local.slice(8, 10);
  return `0${first3} *** ** ${last2}`;
}

/** wa.me linki için: ülke koduyla birlikte, boşluksuz uluslararası format (ör. 905551234522). */
export function toWhatsAppPhone(raw: string): string {
  const local = normalizeTurkishPhone(raw);
  return `90${local}`;
}